packages/engine/tests/simulations/experiment.rs

```diff
@@ -0,0 +1,269 @@
+use std::{
+    fs::{self, File},
+    io::BufReader,
+    path::{Path, PathBuf},
+    sync::Once,
+};
+
+use ::error::{ensure, Report, Result, ResultExt};
+use hash_engine::utils::OutputFormat;
+use regex::Regex;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use serde_json::Value;
+
+use super::error::{ProcessContext, StatusCode, Stderr, Stdout};
+
+#[derive(Debug, Clone)]
+pub enum ExperimentType {
+    SingleRun { num_runs: u64 },
+    Simple { experiment_name: String },
+}
+
+fn parse_file<T: DeserializeOwned>(path: impl AsRef<Path>) -> Result<T> {
+    let path = path.as_ref();
+    serde_json::from_reader(BufReader::new(
+        File::open(path).wrap_err_lazy(|| format!("Could not open file {path:?}"))?,
+    ))
+    .wrap_err_lazy(|| format!("Could not parse {path:?}"))
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct ExperimentOutput {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub json_state: Option<Value>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub globals: Option<Value>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub analysis_outputs: Option<Value>,
+}
+
+fn assert_subset_object(subset: &Value, superset: &Value, path: String) {
```

Comment:
This definitely needs a docstring

Reply:
And honestly probably a unit-test (appreciate that might seem a bit silly considering these are integration tests but it's complicated enough that I think it'd be good to have confidence in it)

Reply:
I will probably just rewrite the function a bit to return `bool` instead. Adding a few tests shouldn't be hard, but I won't aim for a 100% branch coverage.

Reply:
Also the function is now way simpler.